Automattic/wp-calypso
# Post Editor: move post autosave action and isAutosaving flag to Redux

This patch moves the `isAutosaving` flag from the Flux store to Redux. After this change, the `autosave` action only ever touches Redux state: the preview URL and the `isAutosaving` flag. Let's convert it to a Redux thunk then.

This patch also removes the specialized `EDITOR_AUTOSAVE_{SET,RESET}_PREVIEW_URL` actions and merges them into more generic `EDITOR_AUTOSAVE_RESET` and `EDITOR_AUTOSAVE_SUCCESS`.

**How to test:**
This patch touches autosave of an already published post. Test that it still works as expected: after title or content change, the post gets autosaved and when you reopen the editor, restore from autosave will be offered.

Test also autosave of a draft post, which is a regular save. Does it still work?

Autosave interacts with preview. Type something into the title/content area and click "Preview" very quickly, before the throttled autosave starts. Check that the preview window is blank while the autosave is in progress and that the preview that eventually appears contains the title/content edits: these are present only in the autosaved version, not in the published one.
